10.1/root/usr/share/container-scripts/mysql/common.sh

```diff
@@ -162,3 +165,63 @@ function wait_for_mysql_master() {
     sleep 1
   done
 }
+
+# Converts string version to the integer format (5.5 is converted to 505,
+# 10.1 is converted into 1001, etc.
+function version2number() {
+  local version=0
+  local version_text="$1"
+  local version_major=$(echo "$version_text" | sed -e 's/\([0-9]*\)\.\([0-9]*\)\..*$/\1/')
+  local version_minor=$(echo "$version_text" | sed -e 's/\([0-9]*\)\.\([0-9]*\)\..*$/\2/')
+  if [[ $version_major =~ ^[0-9]+$ ]] && [[ $version_minor =~ ^[0-9]+$ ]] ; then
+    version=$((version_major*100+version_minor))
+  fi
+  if [ "$version" -eq 0 ] ; then
+    echo "Error: version '$version_text' cannot be converted to the number."
+    echo "0"
+  fi
+  echo "${version}"
```

Comment:
This is personal preference, but I can't resist -> I really like this pattern:
```bash
versionNumber() {
# Do something...
# and then set global variable $<function_name>_result (versionNumber_result)
printf -v versionNumber_result %d%02d ${1%%.*} ${1##*.}
}
```
By this pattern, you avoid additional shell forks and error prone statements like:
```bash
variable=`method with params`
```